sos/plugins/peripety.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2018 Red Hat, Inc., Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin
+from re import match
+import glob
+
+
+class Peripety(Plugin, RedHatPlugin):
+    '''Peripety Storage Event Monitor'''
+
+    packages = ('peripety',)
+    services = ('peripetyd',)
+
+    def setup(self):
+        self.add_copy_spec('/etc/peripety.conf')
+
+        forbid_reg = [
+            'vd.*',
+            'sr.*'
+        ]
+
+        disks = filter(lambda x: not any(match(reg, x) for reg in forbid_reg),
+                       [d.split('/')[-1] for d in glob.glob('/sys/block/*')])
```

Comment:
I'd really like to get things like the long-standing ask for disk/network etc. iterators/collections done for this. Then we can lose all these ad-hoc "walk all the disks" type things.

Does this want to be also filtering things like ram disks, loopbacks etc.?

Reply:
I wasn't sure if we were still looking at those iterators from a previous conversation we had on IRC. If those are still on the wish list I agree that'd be a big win to get something hammered out there.

I _think_ loopbacks should be fine, I'm not sure on ram disks. Will test it out.

Reply:
Tried loopbacks and ramdisks, both fail so I've added them to the forbids list.

peripety is still pretty new so I imagine these will be tweaked a decent amount going forward.